docker/convert.go

```diff
@@ -56,11 +56,13 @@ func ConvertToAPI(s *Service) (*ConfigWrapper, error) {
 func volumes(c *project.ServiceConfig, ctx project.Context) map[string]struct{} {
 	volumes := make(map[string]struct{}, len(c.Volumes))
 	for k, v := range c.Volumes {
-		vol := ctx.ResourceLookup.ResolvePath(v, ctx.ComposeFiles[0])
+		if len(ctx.ComposeFiles) > 0 {
```

Comment:
The volume path may still be absolute, even if a Compose file is given. `RevolvePath` will check if the path is absolute first, and if it is then it'll just return the given path.
https://github.com/docker/libcompose/blob/master/lookup/file.go#L30